### PR TITLE
[exporter/splunkhec] Skip datapoint.Sum NaN values

### LINKED ISSUE #14877: [exporter/splunkhec] dropped metric event because of unsupported value: NaN
## Description

splunkhec exporter  throws `unsupported value: NaN` error when `SummaryDataPoint.Sum` and `HistogramDataPoint.Sum` returns `NaN` during json marshal

## Expected Result

It should marshal without any error

## Actual Result

```
Permanent error: dropped metric event: &{0xc030652cd0 aks-agentpool-34656813-vmss_9 kubernetes  aks_metrics_fni metric map[azure.resourcegroup.name:XYZ azure.vm.name:aks-agentpool-34656813-vmss_9 azure.vm.scaleset.name:aks-agentpool-34656813-vmss azure.vm.size:Standard_D8as_v5 cloud.account.id:ID cloud.platform:azure_aks cloud.provider:azure cloud.region:westeurope deployment.environment:qas host.id:8bb51409-5e87-44db-95d5-4781e00700bd http.scheme:http k8s.cluster.name:cluster k8s.namespace.name:istio-system k8s.node.name:aks-agentpool-34656813-vmss000009 k8s.pod.name:istio-ingressgateway-internal-7dd4b7f89d-5vgfj k8s.pod.uid:fcfeecd1-6b94-44c4-af8e-8b13a59deb9b metric_name:istio_agent_go_sched_latencies_seconds_sum:NaN metric_type:Histogram net.host.name:HOST net.host.port:15020 os.type:linux service.instance.id:HOST:15020 service.name:prometheus_simple/HOST:15020]}, error: splunk.Event.Fields: unsupported value: NaN
```

### Collector version

v0.61.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hvaghani221**
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14877

`SummaryDataPoint.Sum` and `HistogramDataPoint.Sum` method may return `NaN` value. It may cause an error during JSON marshal

### CODE REVIEW

**Comment by harshit-splunk** [comment]
Not sure why tests are failing!

File: exporter/splunkhecexporter/metricdata_to_splunk.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func mapMetricToSplunkEvent(res pcommon.Resource, m pmetric.Metric, config *Conf
 			{
 				fields := cloneMap(commonFields)
 				populateAttributes(fields, dataPt.Attributes())
-				fields[metricFieldName+sumSuffix] = dataPt.Sum()
+				fields[metricFieldName+sumSuffix] = sanitizeFloat(dataPt.Sum())
```
Comment (bogdandrutu) [inline exporter/splunkhecexporter/metricdata_to_splunk.go:121]: In general should not be NaN (may be if dataPt.Flags().NoRecordedValue() == true), check also for "HasSum()" (it is 0 if not present).

Comment (harshit-splunk) [reply exporter/splunkhecexporter/metricdata_to_splunk.go:121]: It is throwing the following errors in the customer environment. 
Error: 
```
Permanent error: dropped metric event: &{0xc030652cd0 aks-agentpool-34656813-vmss_9 kubernetes  aks_metrics_fni metric map[azure.resourcegroup.name:XYZ azure.vm.name:aks-agentpool-34656813-vmss_9 azure.vm.scaleset.name:aks-agentpool-34656813-vmss azure.vm.size:Standard_D8as_v5 cloud.account.id:ID cloud.platform:azure_aks cloud.provider:azure cloud.region:westeurope deployment.environment:qas host.id:8bb51409-5e87-44db-95d5-4781e00700bd http.scheme:http k8s.cluster.name:cluster k8s.namespace.name:istio-system k8s.node.name:aks-agentpool-34656813-vmss000009 k8s.pod.name:istio-ingressgateway-internal-7dd4b7f89d-5vgfj k8s.pod.uid:fcfeecd1-6b94-44c4-af8e-8b13a59deb9b metric_name:istio_agent_go_sched_latencies_seconds_sum:NaN metric_type:Histogram net.host.name:HOST net.host.port:15020 os.type:linux service.instance.id:HOST:15020 service.name:prometheus_simple/HOST:15020]}, error: splunk.Event.Fields: unsupported value: NaN
```

Comment (hailpis1) [reply exporter/splunkhecexporter/metricdata_to_splunk.go:121]: Customer here. Let me know if you need any additional analysis/logs/whatever.

Comment (dmitryax) [reply exporter/splunkhecexporter/metricdata_to_splunk.go:121]: The HEC field shouldn't be reported if the sum isn't set. That can be checked with `dataPt.HasSum()`, adding that condition should solve the problem.

Comment (harshit-splunk) [reply exporter/splunkhecexporter/metricdata_to_splunk.go:121]: Sorry for the silly question, I haven't worked with metric data. Should I only skip the sum metric(count, or every metric related to the data point if `dataPt.HasSum()` returns false?

Comment (dmitryax) [reply exporter/splunkhecexporter/metricdata_to_splunk.go:121]: > Sorry for the silly question, I haven't worked with metric data. Should I only skip the sum metric or every metric related to the data point if dataPt.HasSum() returns false?

Yes, please

Comment (harshit-splunk) [reply exporter/splunkhecexporter/metricdata_to_splunk.go]: Thanks for your input. `dataPt.HasSum()` returns true even if sum in `NaN`. So, I have added validation for `NaN` for the sum.

Comment (harshit-splunk) [reply exporter/splunkhecexporter/metricdata_to_splunk.go]: Thanks for your input. `dataPt.HasSum()` returns true even if sum in `NaN`. So, I have added validation for `NaN` for the sum.

File: exporter/splunkhecexporter/metricdata_to_splunk.go [hunk 2]
```diff
@@ -182,7 +182,7 @@ func mapMetricToSplunkEvent(res pcommon.Resource, m pmetric.Metric, config *Conf
 		for gi := 0; gi < pts.Len(); gi++ {
 			dataPt := pts.At(gi)
 			// first, add one event for sum, and one for count
-			{
+			if !math.IsNaN(dataPt.Sum()) {
```
Comment (dmitryax) [inline exporter/splunkhecexporter/metricdata_to_splunk.go:185]: Why without `dataPt.HasSum()` here?

File: exporter/splunkhecexporter/metricdata_to_splunk.go (hunk 1 above)
Comment (dmitryax) [reply exporter/splunkhecexporter/metricdata_to_splunk.go]: Which input/receiver do you use that produce `NaN` sums?

Also what about `dataPt.Flags().NoRecordedValue()` ? Is that flag set to true?

Comment (harshit-splunk) [reply exporter/splunkhecexporter/metricdata_to_splunk.go]: > Which input/receiver do you use that produces NaN sums?

From the logs, it looks like `prometheus_simple` is producing that metric.
> Also what about dataPt.Flags().NoRecordedValue() ? Is that flag set to true?

I haven't checked it yet.

Comment (harshit-splunk) [reply exporter/splunkhecexporter/metricdata_to_splunk.go]: @hailpis1 can help us in reproducing the issue. 
From my understanding, `NaN` sum is coming from the scraping endpoint. And Prometheus receiver is collecting it as it is. What is expected from the Prometheus receiver when it receives such metrics?

Meanwhile, if the current changes look good, can we merge them? @hailpis1 is waiting on this.

File: .chloggen/splunkhecexporter-sanitize-NaN-values.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: splunkhecexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Santitized value returned from `SummaryDataPoint.Sum` and `HistogramDataPoint.Sum` as it may return `NaN` value."
```
Comment (dmitryax) [inline .chloggen/splunkhecexporter-sanitize-NaN-values.yaml:8]:
```suggestion
note: "Skip `SummaryDataPoint.Sum` and `HistogramDataPoint.Sum` NaN values"
```

**Review by dmitryax — APPROVED** [review]

File: exporter/splunkhecexporter/metricdata_to_splunk.go (hunk 1 above)
Comment (dmitryax) [reply exporter/splunkhecexporter/metricdata_to_splunk.go]: Sure it looks good to me

**Pull request merged**